api_types.ts

```diff
@@ -32,6 +32,12 @@ export interface ArrayType<T extends Type> {
   type: 'array';
   /** The type of the items in this array. */
   items: T;
+  /** If true, this array will accept empty values as `undefined` for non-string types. */
```

Comment:
Interesting, is the idea that for strings we don't have to support sparse arrays since we'll coerce to the empty string? So that means StringArray, HtmlArray, ImageArray, FileArray don't need to be sparse? And boolean could similarly coerce to false. So that just means it's probably just NumberArray and DateArray that require declaring params as sparse?

I don't object, but the inconsistency between array types is a bit odd and I wonder if developers will have a hard time with it. As usual, mostly deferring to @ekoleda-codaio about whether consistency is better or avoiding sparseness when not necessary is better.

Reply:
oh I think HtmlArray, ImageArray and FileArray would still be sparse. (just think about an empty column)

I exempted non-string value because an empty string column still still a valid string. while empty image/file column should undefined.

Reply:
Ok yeah that seems reasonable.

I guess it's still a little weird why the same blank cell results in undefined for HtmlArray but `''` for a StringArray, but maybe coercion isn't that hard for people to grasp. Even if we typed StringArray to be sparse, I guess CFL is already coercing blanks to strings so we'd never see `undefined` in practice?

@ekoleda-codaio WDYT, should StringArray be a special snowflake that doesn't need `undefined`, or should it be consistent with all the rest?

Reply:
thinking more about this I guess all array types can be undefined. essentially if the input value is not coerce'able for the given type it would be passed in as undefined. 

so if we pass an image column into a string array (assuming coerce doesn't work in this case), I guess it should be undefined?

Reply:
but empty text column should be just `''` for string array I think. it makes sense to most cases I think.

Reply:
That makes sense that it depends on the coercability of the source data type 👍

Reply:
I'll proceed with this direction, and finish up all the remaining data types.

Reply:
I think ideally we would make all array types work the same. Having an empty cell result in an empty string for `StringArray` but undefined for `NumberArray` is unexpected, and exposes an implementation detail of Coda. That said, if it's too hard to change I think we could document the inconsistency any be fine.